Userland/Services/WindowServer/Menubar.h

```diff
@@ -18,7 +18,20 @@ class Menubar {
 public:
     void add_menu(Menu& menu, Gfx::IntRect window_rect)
     {
-        // FIXME: Check against duplicate menu additions.
+        bool duplicate_menu_detected = false;
+
+        for_each_menu([&](Menu& existing_menu) {
+            if (&menu == &existing_menu) {
+                dbgln("Duplicate Menu \"{}\" ({})", menu.name(), &menu);
+                duplicate_menu_detected = true;
+                return IterationDecision::Break;
+            }
+            return IterationDecision::Continue;
+        });
```

Comment:
In my opinion it's unlikely that we need to frequently add new menus so long term tracking and holding more memory isn't justified, even if it's very small. This memory usage is temporary and I don't see a use case that is adding enough menu items that this will be significantly slower but I am open to examples which this should be considered and improved later on. This at least resolves the issue for now. continous improvement > delayed perfection :)

Reply:
Fair enough! :^)